From 1bdde19d88e5c3273d828bbc7e667a1ca80a245f Mon Sep 17 00:00:00 2001
Message-Id: <1bdde19d88e5c3273d828bbc7e667a1ca80a245f.1349175436.git.minovotn@redhat.com>
In-Reply-To: <94968b7fa9b14e71f004474d7ce77e189e6a2bf3.1349175436.git.minovotn@redhat.com>
References: <94968b7fa9b14e71f004474d7ce77e189e6a2bf3.1349175436.git.minovotn@redhat.com>
From: Eduardo Habkost <ehabkost@redhat.com>
Date: Fri, 28 Sep 2012 16:50:13 +0200
Subject: [PATCH 09/34] target-i386: Drop unused setscalar() macro
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

RH-Author: Eduardo Habkost <ehabkost@redhat.com>
Message-id: <1348851023-31907-9-git-send-email-ehabkost@redhat.com>
Patchwork-id: 42518
O-Subject: [RHEL6 qemu-kvm PATCH 08/18] target-i386: Drop unused setscalar() macro
Bugzilla: 833152
RH-Acked-by: Alex Williamson <alex.williamson@redhat.com>
RH-Acked-by: Paolo Bonzini <pbonzini@redhat.com>
RH-Acked-by: Igor Mammedov <imammedo@redhat.com>
RH-Acked-by: Orit Wasserman <owasserm@redhat.com>

From: Andreas Färber <afaerber@suse.de>

Bugzilla: 833152

It was only used by now removed setfeatures() function.

Suggested-by: Igor Mammedov <imammedo@redhat.com>
Signed-off-by: Andreas Färber <afaerber@suse.de>
(cherry picked from commit 473955e5c3bce09c007ba3b64937cfca6f18f525)

Conflicts:
	target-i386/cpuid.c

(Simple conflict because cpu_clear_apic_feature() is not present on
RHEL-6).

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 target-i386/cpuid.c | 14 --------------
 1 file changed, 14 deletions(-)

Signed-off-by: Michal Novotny <minovotn@redhat.com>
---
 target-i386/cpuid.c | 14 --------------
 1 file changed, 14 deletions(-)

diff --git a/target-i386/cpuid.c b/target-i386/cpuid.c
index 3e15176..c18f9b2 100644
--- a/target-i386/cpuid.c
+++ b/target-i386/cpuid.c
@@ -1055,20 +1055,6 @@ int cpu_x86_register (CPUX86State *env, const char *cpu_model)
     return 0;
 }
 
-#if !defined(CONFIG_LINUX_USER)
-/* interpret radix and convert from string to arbitrary scalar,
- * otherwise flag failure
- */
-#define setscalar(pval, str, perr)                      \
-{                                                       \
-    char *pend;                                         \
-    unsigned long ul;                                   \
-                                                        \
-    ul = strtoul(str, &pend, 0);                        \
-    *str && !*pend ? (*pval = ul) : (*perr = 1);        \
-}
-#endif	/* !CONFIG_LINUX_USER */
-
 /* Initialize list of CPU models, filling some non-static fields if necessary
  */
 void x86_cpudef_setup(void)
-- 
1.7.11.4

